### PR TITLE
Fix 991: Custom code style for xml

### PR DESCRIPTION
**Pull request by nikitamarysolomanpvt**
## Explanation
This PR updates .idea files (by importing custom project code style)to avoid rearranging of tags whenever code reformatter is used.

## Documentation 

https://docs.google.com/document/d/1AHIJagMfATgpbNCFmYoYaSqzsKp8JjTgPLQFXEFJdUA/edit?usp=sharing

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by nikitamarysolomanpvt** [comment]
> ## Documentation
> https://docs.google.com/document/d/1AHIJagMfATgpbNCFmYoYaSqzsKp8JjTgPLQFXEFJdUA/edit?usp=sharing
> 


Please suggest changes if required as the document provided will be added in wiki once approved.

**Review by rt4914 — APPROVED** [review]
Works correctly for me.

**Review by veena14cs — APPROVED** [review]
Works for me.

**Comment by BenHenning** [comment]
I’m unfortunately not able to understand what the proposed style here is from the codeStyle file alone. :) Can you add some example XML layouts before and after this change to show what it specifically being changed here? Is all this fixing is the reordering issue or something more?

Also adding Vinita to take a look.

**Comment by nikitamarysolomanpvt** [comment]
> I’m unfortunately not able to understand what the proposed style here is from the codeStyle file alone. :) Can you add some example XML layouts before and after this change to show what it specifically being changed here? Is all this fixing is the reordering issue or something more?
> 
> Also adding Vinita to take a look.

@BenHenning 

**source code**
```
<?xml version="1.0" encoding="utf-8"?>
<layout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto">

    <androidx.constraintlayout.widget.ConstraintLayout
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:orientation="vertical">

        
        <TextView
            android:id="@+id/slide_title_text_view"
            android:layout_width="0dp"
            android:layout_height="wrap_content"
            android:layout_marginStart="48dp"
            android:layout_marginTop="16dp"
            android:layout_marginEnd="48dp"
            android:layout_marginBottom="16dp"
            android:fontFamily="sans-serif-medium"
            android:gravity="center_horizontal"
            android:maxLines="1"
            android:text="@{viewModel.title}"
            android:textColor="@color/oppiaPrimaryText"
            android:textSize="24sp"
            app:layout_constraintBottom_toTopOf="@id/slide_description_text_view"
            app:layout_constraintEnd_toEndOf="parent"
            app:layout_constraintStart_toStartOf="parent" />
      
        <ImageView
            android:id="@+id/slide_image_view"
            android:layout_width="0dp"
            android:layout_height="0dp"
            android:contentDescription="@{viewModel.contentDescription}"
            android:src="@{viewModel.slideImage}"
            app:layout_constraintDimensionRatio="V,4:4"
            app:layout_constraintEnd_toEndOf="parent"
            app:layout_constraintStart_toStartOf="parent"
            app:layout_constraintTop_toTopOf="parent" />

        <TextView
            android:id="@+id/slide_description_text_view"
            android:layout_width="0dp"
            android:layout_height="wrap_content"
            android:layout_marginStart="48dp"
            android:layout_marginEnd="48dp"
            android:layout_marginBottom="32dp"
            android:fontFamily="sans-serif"
            android:gravity="center_horizontal"
            android:maxLines="2"
            android:minLines="2"
            android:text="@{viewModel.description}"
            android:textColor="@color/oppiaPrimaryText"
            android:textSize="16sp"
            app:layout_constraintBottom_toBottomOf="parent"
            app:layout_constraintEnd_toEndOf="parent"
            app:layout_constraintStart_toStartOf="parent" />
    </androidx.constraintlayout.widget.ConstraintLayout>

    <data>

        <variable
            name="viewModel"
            type="org.oppia.app.onboarding.OnboardingSlideViewModel" />
    </data>
</layout>

```
**Current style**
```
<?xml version="1.0" encoding="utf-8"?>
<layout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto">

    <androidx.constraintlayout.widget.ConstraintLayout
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:orientation="vertical">

        <ImageView
            android:id="@+id/slide_image_view"
            android:layout_width="0dp"
            android:layout_height="0dp"
            android:contentDescription="@{viewModel.contentDescription}"
            android:src="@{viewModel.slideImage}"
            app:layout_constraintDimensionRatio="V,4:4"
            app:layout_constraintEnd_toEndOf="parent"
            app:layout_constraintStart_toStartOf="parent"
            app:layout_constraintTop_toTopOf="parent" />

        <TextView
            android:id="@+id/slide_title_text_view"
            android:layout_width="0dp"
            android:layout_height="wrap_content"
            android:layout_marginStart="48dp"
            android:layout_marginTop="16dp"
            android:layout_marginEnd="48dp"
            android:layout_marginBottom="16dp"
            android:fontFamily="sans-serif-medium"
            android:gravity="center_horizontal"
            android:maxLines="1"
            android:text="@{viewModel.title}"
            android:textColor="@color/oppiaPrimaryText"
            android:textSize="24sp"
            app:layout_constraintBottom_toTopOf="@id/slide_description_text_view"
            app:layout_constraintEnd_toEndOf="parent"
            app:layout_constraintStart_toStartOf="parent" />

        <TextView
            android:id="@+id/slide_description_text_view"
            android:layout_width="0dp"
            android:layout_height="wrap_content"
            android:layout_marginStart="48dp"
            android:layout_marginEnd="48dp"
            android:layout_marginBottom="32dp"
            android:fontFamily="sans-serif"
            android:gravity="center_horizontal"
            android:maxLines="2"
            android:minLines="2"
            android:text="@{viewModel.description}"
            android:textColor="@color/oppiaPrimaryText"
            android:textSize="16sp"
            app:layout_constraintBottom_toBottomOf="parent"
            app:layout_constraintEnd_toEndOf="parent"
            app:layout_constraintStart_toStartOf="parent" />
    </androidx.constraintlayout.widget.ConstraintLayout>

    <data>

        <variable
            name="viewModel"
            type="org.oppia.app.onboarding.OnboardingSlideViewModel" />
    </data>
</layout>

```
**After applying custom style**
```
<?xml version="1.0" encoding="utf-8"?>
<layout xmlns:android="http://schemas.android.com/apk/res/android"
  xmlns:app="http://schemas.android.com/apk/res-auto">

  <data>

    <variable
      name="viewModel"
      type="org.oppia.app.onboarding.OnboardingSlideViewModel" />
  </data>

  <androidx.constraintlayout.widget.ConstraintLayout
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:orientation="vertical">

    <ImageView
      android:id="@+id/slide_image_view"
      android:layout_width="0dp"
      android:layout_height="0dp"
      android:contentDescription="@{viewModel.contentDescription}"
      android:src="@{viewModel.slideImage}"
      app:layout_constraintDimensionRatio="V,4:4"
      app:layout_constraintEnd_toEndOf="parent"
      app:layout_constraintStart_toStartOf="parent"
      app:layout_constraintTop_toTopOf="parent" />

    <TextView
      android:id="@+id/slide_title_text_view"
      android:layout_width="0dp"
      android:layout_height="wrap_content"
      android:layout_marginStart="48dp"
      android:layout_marginTop="16dp"
      android:layout_marginEnd="48dp"
      android:layout_marginBottom="16dp"
      android:fontFamily="sans-serif-medium"
      android:gravity="center_horizontal"
      android:maxLines="1"
      android:text="@{viewModel.title}"
      android:textColor="@color/oppiaPrimaryText"
      android:textSize="24sp"
      app:layout_constraintBottom_toTopOf="@id/slide_description_text_view"
      app:layout_constraintEnd_toEndOf="parent"
      app:layout_constraintStart_toStartOf="parent" />

    <TextView
      android:id="@+id/slide_description_text_view"
      android:layout_width="0dp"
      android:layout_height="wrap_content"
      android:layout_marginStart="48dp"
      android:layout_marginEnd="48dp"
      android:layout_marginBottom="32dp"
      android:fontFamily="sans-serif"
      android:gravity="center_horizontal"
      android:maxLines="2"
      android:minLines="2"
      android:text="@{viewModel.description}"
      android:textColor="@color/oppiaPrimaryText"
      android:textSize="16sp"
      app:layout_constraintBottom_toBottomOf="parent"
      app:layout_constraintEnd_toEndOf="parent"
      app:layout_constraintStart_toStartOf="parent" />
  </androidx.constraintlayout.widget.ConstraintLayout>
</layout>

```

File: .idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -1,16 +1,14 @@
 <component name="ProjectCodeStyleConfiguration">
   <code_scheme name="Project" version="173">
-    <JavaCodeStyleSettings>
```
Comment (BenHenning) [inline .idea/codeStyles/Project.xml]: Why remove these? We want to avoid wildcard imports per the style guide. Or were these not doing anything?

**Review by BenHenning — APPROVED** [review]
Thanks for the additional context @nikitamarysolomanpvt. The changes LGTM minus the code style ones. Please address that comment before submitting.

File: .idea/codeStyles/Project.xml (hunk 1 above)
Comment (nikitamarysolomanpvt) [reply .idea/codeStyles/Project.xml]: @BenHenning Previously we where not using this. i.e default value is sufficient i.e. 5... If you insist we can use this JavaCodeStyleSettings customisations. Also we are not entertaining  wildcard imports isn’t ?

**Review by vinitamurthi — APPROVED** [review]

File: .idea/codeStyles/Project.xml (hunk 1 above)
Comment (vinitamurthi) [reply .idea/codeStyles/Project.xml]: My only concern about using default value is that it may change for different android studio versions

Comment (BenHenning) [reply .idea/codeStyles/Project.xml]: I agree with Vinita--is it possible for us to specify a default? In general, we never want the automatic wildcard support. And no, we shouldn't be using wildcard imports ever per https://developer.android.com/kotlin/style-guide#import_statements. :)

Comment (nikitamarysolomanpvt) [reply .idea/codeStyles/Project.xml]: Okay will do that

**Pull request merged**